src/wallet/test/fuzz/coinselection.cpp

```diff
@@ -101,16 +101,23 @@ FUZZ_TARGET(coinselection)
 
     // Run coinselection algorithms
     auto result_bnb = SelectCoinsBnB(group_pos, target, cost_of_change, MAX_STANDARD_TX_WEIGHT);
+    if (result_bnb) {
+        (void)result_bnb->GetShuffledInputVector();
+    }
```

Comment:
Couldn't we assert that BnB `GetChange()` is always 0?

Reply:
Does it worth? BnB internally calls `ComputeSetAndWaste` which checks change. Any unexpected action regarding it would affect the assertion - `assert(best_waste == result.GetWaste());`.

Reply:
I think that it worth. One of the fuzzing goals is to assert expected behavior externally. Black-box style. So if the internal implementation changes by mistake (or if there is a way to by-pass the expected behavior that we aren't aware of), the fuzzer catches it.

Also, the internal BnB `ComputeSetAndWaste` call checks that the final selection is within a certain range: `target <= value <  target + change cost`. It is indirectly checking `GetChange()` result but I don't think that we should rely on that. If `change_cost` is also 0, the `GetChange` result doesn't really matter, the same code will be executed in both branches of the `ComputeAndSetWaste` function.

Reply:
Fair enough, gonna address it.

Reply:
Done!